scripts/build/solver-stp.sh

```diff
@@ -4,34 +4,30 @@
 
 # Make sure we exit if there is a failure
 set -e
+DIR=$(dirname "$(readlink -f "$0" || stat -f "$0")")
+source "${DIR}/common-defaults.sh"
 
-STP_LOG="$(pwd)/stp-build.log"
-
+cd ${BASE}
 if [ "x${STP_VERSION}" != "x" ]; then
-  # Get Sanitizer flags
-  source ${KLEE_SRC}/.travis/sanitizer_flags.sh
-
   # Build minisat
-  git clone https://github.com/stp/minisat
-  cd minisat
-  mkdir build
-  cd build
-  MINISAT_DIR=`pwd`
-  if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+  git_clone_or_update https://github.com/stp/minisat "${BASE}/minisat"
```

Comment:
What's wrong with just a simple clone? No need to make everything complicated, because someone might use this outside docker and it will probably be you anyway.

Reply:
Well, the problem is that everybody has her/his own script to do it and starts again and again doing it.
The idea is to get rid of it and make it accessible (with and without docker).

If you're just using one simple setup - fine. But I guess there are people out there (beyond me), wo have more complex requirements. (Different LLVM versions, sanitizer builds, ...)

Reply:
I think if we want to support other people using these scripts, they should be well documented and have some tests too on various distributions and setups.

Reply:
The idea is to provide scripts that are the same for:
* your local builds
* docker builds
* travis builds

As they are tested with travis - they are more likely to work with other systems as well.
Yes, documentation is still missing.